### PR TITLE
MAINTAINERS: Update renesas-rcar platforms

### PR DESCRIPTION
**Pull request by aaillet**
New `rcar` folder for renesas dts files:
We needed to move rcar arm dts files to a new `rcar` folder to better delimit renesas ranges (after smartbond support creation).

MAINTAINERS file update for renesas r-car:
Having two renesas r-car area is not the better way to notify interested users, so we are merging these.
We are also updating collaborators depending on people working on the subject right now

### CODE REVIEW

**Comment by aaillet** [comment]
ping @xakep-amatop

**Comment by xakep-amatop** [comment]
> ping @xakep-amatop

I'm okay with being a collaborator of Renesas R-Car boards and drivers.

**Review by xakep-amatop — APPROVED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@aaillet could you rebase please?

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -2409,34 +2409,25 @@ Renesas R-Car Platforms:
   status: maintained
   maintainers:
     - aaillet
+    - lorc
```
Comment (nashif) [inline MAINTAINERS.yml:2412]: This handle is new to the project, so nothing will be assigned to @lorc etc.  need to go though this first:

https://github.com/zephyrproject-rtos/zephyr/issues/new?assignees=&labels=Role+Nomination&projects=&template=006_nomination.md&title=

**Review by nashif — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml (hunk 1 above)
Comment (lorc) [reply MAINTAINERS.yml:2412]: Done: https://github.com/zephyrproject-rtos/zephyr/issues/61574

**Review by nashif — DISMISSED** [review]

**Comment by nashif** [comment]
btw, title of this PR is misleading, it is not only a MAINTAINER file change, you have DTS related changes in there.

**Comment by aaillet** [comment]
> btw, title of this PR is misleading, it is not only a MAINTAINER file change, you have DTS related changes in there.

Yes, changes are well delimited in two separate commits but as changes are all done in order to better delimit MAINTAINERS area I though it was the right name to use.
Sorry

**Comment by aaillet** [comment]
Just rebased and resolved conflicts !

**Review by pmarzin — APPROVED** [review]
Changes are ok for me. Thanks @aaillet

**Review by fabiobaltieri — APPROVED** [review]

**Review by lorc — APPROVED** [review]

**Comment by kartben** [comment]
@lorc feel free to claim your Zephyr Maintainer badge! https://forms.gle/oCw9iAPLhUsHTapc8
Thanks for stepping up as a maintainer.

**Pull request merged**